### ISSUE TITLE
Improve the git container class

### ISSUE THREAD

**Issue by mapaiva** [mapaiva/GitPie#4]
With the growing of GitPie the git wrapper (The class that performs the pure git commands) become out of pattern and bad documented.

It needs to create a pattern for all the methods signature. For example:

``` javascript
Git.prototype.getCommitHistory = function (opts, callback) { ...

// OR

Git.prototype.getCommitHistory = function (path, opts) {
```

And improve the JSDocs for all the methods too. 

@IonicaBizau Do you thing you can do that? :smile: It will help let the code more documented and organized.

**Comment by IonicaBizau** [comment]
Can we use [`gry`](https://github.com/IonicaBizau/node-gry) there? It would simplify the things a little bit.

I will take a look soon.

**Comment by mapaiva** [comment]
I'm not totally against that. But we need execute sync and async commands and replace the existing methods in `Git` class. If `Gry` can do this and can be too fast as native child proccess are, I don't see any bad in change it. I think that [nodegit](https://github.com/nodegit/nodegit) can be a nice alternative too.

**Comment by IonicaBizau** [comment]
> I think that nodegit can be a nice alternative too.

I tried it three times (I guess one time per year), and still don't know how to use that module... Lol. It's just so complicated to use. :smile: 

What's the case for sync execution? `gry` currently supports only async child process executions.

**Comment by mapaiva** [comment]
I didn't know that use it was so complicated kkkk. I use sync executions for `add`, `discartChangesInFile` and `commit`.

I can look a little antiquated but I really love the idea to code everything from the ground and fit your code with your necessary making a clean solution. It not means that I don't like to use betters solutions.

I guess that GitPie have his own git wrapper, is the best way to have all the control of the commands.

**Comment by IonicaBizau** [comment]
I have some news on this: `oargv@3.0.0` supports returning an array: https://github.com/IonicaBizau/node-oargv

This will be integrated in the `gry` package with `spawn` support. :sparkle:

**Comment by mapaiva** [comment]
Hmmm it's very interesting. Handle flags and arguments in the git wrapper is a mess on GitPie. You can branch and test it with no problems. Even me will take a look sooner :smiley:

**Comment by IonicaBizau** [comment]
Finally, released [`gry@4.1.0`](https://github.com/IonicaBizau/node-gry) which supports both exec and spawn. Probably we only need spawn here. :tada:

**Comment by mapaiva** [comment]
Nice! If `gry` can make perform sync and async tasks, feel free to branch and test it :smile:

**Comment by IonicaBizau** [comment]
@mapaiva What's the use-case of sync tasks? :thought_balloon: Currently it can perform only async operations.

**Comment by mapaiva** [comment]
@IonicaBizau when the user select the files to commit and click on the **COMMIT** button, I synchronously add file per file. I do that, because I tried it in the async way, and with multiple files being added, git raised an error of process concurrency. :/

**Comment by IonicaBizau** [comment]
Well, I can fix that too. You just need to use callbacks. I will take care of that. :smile:

**Comment by mapaiva** [comment]
Wonderfull notice!! LOL :smile:

**Comment by mapaiva** [comment]
With a bug in electron we need to start use promisses to syncronized executions or try another work around #23

**Comment by IonicaBizau** [comment]
@mapaiva Ohhhh... I'm currently busy with random projects, so I'm going to self-unassign. :joy: Sorry, I have no time to solve this, but I will keep an eye on this project development.

**Comment by mapaiva** [comment]
@IonicaBizau ok no problem. :+1:

**Comment by mapaiva** [comment]
Fixed by #34
